File: server/src/main/java/com/backend/server/dto/ChatMessageResponse.java
package com.backend.server.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class ChatMessageResponse {

    private Long id;
    private Long conversationId;
    private Long senderId;
    /** Plain text (decrypted for API responses). */
    private String body;
    private LocalDateTime sentAt;
    /** True if sender is the current user. */
    private boolean fromMe;
}

File: app/src/main/java/com/example/client/data/model/ChatMessageResponse.kt
package com.example.client.data.model

import com.google.gson.annotations.SerializedName

data class ChatMessageResponse(
    @SerializedName("id") val id: Long,
    @SerializedName("conversationId") val conversationId: Long,
    @SerializedName("senderId") val senderId: Long,
    @SerializedName("body") val body: String,
    @SerializedName("sentAt") val sentAt: String,
    @SerializedName("fromMe") val fromMe: Boolean
)
